ijkmedia/ijkplayer/ff_ffplay.c

```diff
@@ -1774,7 +1857,45 @@ static int video_thread(void *arg)
     return ret;
 }
 
-// FFP_MERGE: subtitle_thread
+static int subtitle_thread(void *arg)
+{
+    FFPlayer *ffp = arg;
+    VideoState *is = ffp->is;
+    Frame *sp;
+    int got_subtitle;
+    double pts;
+
+    for (;;) {
+        if (!(sp = frame_queue_peek_writable(&is->subpq)))
+            return 0;
+
+        if ((got_subtitle = decoder_decode_frame(ffp, &is->subdec, NULL, &sp->sub)) < 0)
+            break;
+
+        pts = 0;
+#ifdef FFP_MERGE
+        if (got_subtitle && sp->sub.format == 0) {
+#else
+        if (got_subtitle) {
+#endif
+            if (sp->sub.pts != AV_NOPTS_VALUE)
+                pts = sp->sub.pts / (double)AV_TIME_BASE;
+            sp->pts = pts;
+            sp->serial = is->subdec.pkt_serial;
+            sp->width = is->subdec.avctx->width;
+            sp->height = is->subdec.avctx->height;
+            sp->uploaded = 0;
+
+            /* now we can update the picture count */
+            frame_queue_push(&is->subpq);
+#ifdef FFP_MERGE
+        } else if (got_subtitle) {
+            avsubtitle_free(&sp->sub);
```

Comment:
see line 463 and 1146 of ff_ffplay.c. it frees subtitle when subtitle has been "rendered".